crates/hir_ty/src/autoderef.rs

```diff
@@ -24,8 +24,10 @@ pub fn autoderef<'a>(
     krate: Option<CrateId>,
     ty: InEnvironment<Canonical<Ty>>,
 ) -> impl Iterator<Item = Canonical<Ty>> + 'a {
+    let _p = profile::span("autoderef");
     let InEnvironment { goal: ty, environment } = ty;
     successors(Some(ty), move |ty| {
+        let _p = profile::span("autoderef.step");
         deref(db, krate?, InEnvironment { goal: ty, environment: environment.clone() })
```

Comment:
let's put this into `deref` fn?

Reply:
It's a bit tricky, since this is the declaration: 

https://github.com/rust-analyzer/rust-analyzer/blob/fbe98047d419ade6c87ccf432eb9534393d20020/crates/hir_ty/src/db.rs#L124-L129

I could trace `associated_ty_value_query` more, but isn't it odd that Salsa results are not cached in this particular case?